docs/rules/html-self-closing.md

```diff
@@ -0,0 +1,80 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "svelte/html-self-closing"
+description: "enforce self-closing style"
+---
+
+# svelte/html-self-closing
+
+> enforce self-closing style
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+You can choose either two styles for elements without content
+
+- always: `<div />`
+- never: `<div></div>`
+
+<ESLintCodeBlock fix>
+
+<!-- prettier-ignore-start -->
+<!--eslint-skip-->
+
+```svelte
+<script>
+  /* eslint svelte/html-self-closing: "error" */
+</script>
+
+<!-- ✓ GOOD -->
+<div />
```

Comment:
> `div` is not self-closing in HTML
> 'html-compliant' should be default IMO.

I think we should make the default of `prettier-plugin-svelte` the default behavior rather than the HTML standard. So I think `all` should be the default.
`prettier-plugin-svelte` has long been the default style for the community.
Turn off these rules when users use plugins with `prettier-plugin-svelte`, but I think it makes sense to adopt the same style as `prettier-plugin-svelte` by default.

Reply:
>  we should have three options: 'all' | 'html-compliant' | 'none'

I think the current options can already do the same. Why do you think we need three options?

> ```jsonc
> {
>   "svelte/html-self-closing": [
>     "error",
>     {
>       "void": "always", // or "always" or "ignore"
>       "normal": "always", // or "never" or "ignore"
>       "component": "always", // or "never" or "ignore"
>       "svelte": "always" // or "never" or "ignore"
>     }
>   ]
> }
> ```
> 
> - `void` (`"always"` by default)... Style of HTML void elements
> - `component` (`"always"` by default)... Style of svelte components
> - `svelte` (`"always"` by default)... Style of svelte special elements (`<svelte:head>`, `<svelte:self>`)
> - `normal` (`"always"` by default)... Style of other elements
> 
> Every option can be set to
> - "always" (`<div />`)
> - "never" (`<div></div>`)
> - "ignore" (either `<div />` or `<div></div>`)

Reply:
@ota-meshi See also https://github.com/sveltejs/prettier-plugin-svelte/issues/284

Reply:
Oh. I didn't know that.
Hmm. The maintainers don't seem to be involved in the discussion, so I'm still not sure if they're thinking of doing so by default.

Reply:
For example, I'd prefer `<br />` and `<div></div>`, it seems we can't achieve this for current options?

Reply:
> I think the current options can already do the same. Why do you think we need three options?

Hah, right, it makes sense. Although I still think one option with different values is much easiler to understand and use like prettier https://prettier.io/docs/en/options.html#html-whitespace-sensitivity

Reply:
It's nice to have easier options, but Svelte isn't HTML, so you need to be clear about how each option works.

Reply:
The linked option is not related to tag closing, it's related to whitespace and prettier-plugin-svelte already adheres to that option.

Reply:
> The linked option is not related to tag closing, it's related to whitespace and prettier-plugin-svelte already adheres to that option.

I know, I'm just saying we can have such `single` option maybe.

Reply:
> I know, I'm just saying we can have such `single` option, maybe.

That's a good idea, but I'd also keep the current configuration in overrides section, because there could be someone who only wants self closing on tags in specific category, and it would be waste to just throw away the option to set that.

Reply:
@marekvospel Do you currently have time to work on this PR?

I think the new single option is that we can add features later, so if you don't have time now, after merging this PR, we can work on a new PR when you/we have time.

Reply:
@ota-meshi no, I'll have time in 2 weeks or later. I'll create a new PR when I have time